tensorboard/webapp/plugins/npmi/util/filter_annotations_test.ts

```diff
@@ -0,0 +1,455 @@
+            metric: 'test',
+            run: 'run_1',
+            nPMIValue: 0.5178,
+            countValue: 100,
+          },
+          {
+            annotation: 'annotation_1',
+            metric: 'other',
+            run: 'run_1',
+            nPMIValue: 0.815,
+            countValue: 100,
+          },
+          {
+            annotation: 'annotation_1',
+            metric: 'test',
+            run: 'run_2',
+            nPMIValue: 0.02157,
+            countValue: 101,
+          },
+          {
+            annotation: 'annotation_1',
+            metric: 'other',
+            run: 'run_2',
+            nPMIValue: -0.02157,
+            countValue: 101,
+          },
+        ],
+        annotation_2: [
+          {
+            annotation: 'annotation_2',
+            metric: 'test',
+            run: 'run_1',
+            nPMIValue: null,
+            countValue: 572,
+          },
+          {
+            annotation: 'annotation_2',
+            metric: 'other',
+            run: 'run_1',
+            nPMIValue: -1.0,
+            countValue: 53,
+          },
+        ],
+        annotation_3: [
+          {
+            annotation: 'annotation_3',
+            metric: 'test',
+            run: 'run_1',
+            nPMIValue: 0.757,
+            countValue: 572,
+          },
+          {
+            annotation: 'annotation3',
+            metric: 'other',
+            run: 'run_1',
+            nPMIValue: 0.05,
+            countValue: 53,
+          },
+          {
+            annotation: 'annotation_3',
+            metric: 'test',
+            run: 'run_2',
+            nPMIValue: -0.157,
+            countValue: 572,
+          },
+          {
+            annotation: 'annotation3',
+            metric: 'other',
+            run: 'run_2',
+            nPMIValue: -0.05,
+            countValue: 53,
+          },
+        ],
+      });
+    });
+
+    it('returns empty result if no runs active', () => {
+      const result = filterAnnotations(
+        annotationData,
+        [],
+        metricArithmetic,
+        metricFilters,
+        metrics,
+        ''
+      );
+      expect(result).toEqual({});
+    });
+
+    it('returns empty result if no metrics active', () => {
+      const result = filterAnnotations(
+        annotationData,
+        activeRuns,
+        metricArithmetic,
+        metricFilters,
+        [],
+        ''
+      );
+      expect(result).toEqual({});
+    });
+
+    it('returns empty result if no annotation data', () => {
+      const result = filterAnnotations(
+        {},
+        activeRuns,
+        metricArithmetic,
+        metricFilters,
+        metrics,
+        ''
+      );
+      expect(result).toEqual({});
+    });
+
+    it('returns correct result with annotations regex active', () => {
+      const result = filterAnnotations(
+        annotationData,
+        activeRuns,
+        metricArithmetic,
+        metricFilters,
+        metrics,
+        '1'
```

Comment:
Make sure it is regex with more interesting regex? `1$` or `Ann.+_1$` would do the job.